### DISCUSSION TITLE
possible to use different base image?

### DISCUSSION THREAD

**Discussion by jack-evans** [fluxcd/flux2#2047] [Q&A] [answered]
Currently the controllers use a mix of debian and Alpine. Was wondering how difficult would it be to have them be based on a different base image, e.g. ubi or even scratch? or is there hard requirements on the base OS?

**Comment by hiddeco** [comment, marked as answer]
There are no hard requirements, as in, you can theoretically produce an image which would work as well based on another OS.

We can not provide support for it however, as the requirements around building both the source-controller and image-automation-controller are tricky due to their dependency on C libraries, and depending on how you configure it, some features may break due to OS specific linking or library configuration issues: https://github.com/fluxcd/golang-with-libgit2#rationale

We have the ambition to drop the dependency on C libraries for crypto and transport in an upcoming release (which would only leave `libgit2` itself as a C dependency), allowing us to start preparing for static binaries, but this is at present blocked due to an upstream bug: https://github.com/fluxcd/source-controller/pull/465
